### PR TITLE
tests/main/snap-quota-groups: fix spread test

### PR DESCRIPTION
**Pull request by Meulengracht**
In newer systemd, empty slices can use up to 12KB because of cached memory, and this messed up the memory quota spread test that expects either 0 or 4KB.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you - this is fine but if it changes again (to e.g. 16kb on s390x)  I think we should just check if it's less than 640kb or something like arbitrary (but much bigger) like this.

**Comment by sergiocazzolato** [comment]
@Meulengracht thanks for the fix
I see the snap-quotas test also failing https://github.com/snapcore/spread-cron/runs/5787209941?check_suite_focus=true#step:3:11857 
Could we include the fix also for that test?

File: tests/main/snap-quota-groups/task.yaml [hunk 1]
```diff
@@ -170,17 +170,17 @@ execute: |
...
   # where an empty cgroup has exactly 0, but on newer systems there is some 
-  # minimum amount of accounting memory for an empty cgroup, so it shows up as 
-  # 4K
-  if [ "$snapdSaysMemUsage" != null ] && [ "$snapdSaysMemUsage" != 0 ] && [ "$snapdSaysMemUsage" != 4096 ] ; then
+  # minimum amount of accounting memory for an empty cgroup, which is observed
+  # to be up to 12KiB due to cached memory.
+  if [ "$snapdSaysMemUsage" != null ] && [ "$snapdSaysMemUsage" != 0 ] && [ "$snapdSaysMemUsage" != 4096 ] && [ "$snapdSaysMemUsage" != 12288 ] ; then
     echo "unexpected memory usage for empty quota group group-five, snapd says: $snapdSaysMemUsage"
     exit 1
   fi
```
Comment (bboozzoo) [inline tests/main/snap-quota-groups/task.yaml:180]:
```suggestion
case "$snapdSaysMemUsage" in
    null|0|4096|12288)
        # one of expected values
        ;; 
    *)
        echo "unexpected memory usage for empty quota group group-five, snapd says: $snapdSaysMemUsage"
        exit 1
esac
```

**Review by bboozzoo — COMMENTED** [review]
maybe a little tweak

File: tests/main/snap-quota-groups/task.yaml (hunk 1 above)
Comment (Meulengracht) [reply tests/main/snap-quota-groups/task.yaml:180]: Really nice suggestion, will add this!

**Review by bboozzoo — APPROVED** [review]
LGTM

**Comment by mvo5** [comment]
This now fails with:
```
2022-04-04T09:52:23.4157127Z Checking that all quotas can be listed
2022-04-04T09:52:23.4157505Z + snap quotas
2022-04-04T09:52:23.4157856Z + cat -n
2022-04-04T09:52:23.4158200Z error: memory usage unavailable
2022-04-04T09:52:23.4159279Z + MATCH '     1\s+Quota\s+Parent\s+Constraints\s+Current$'
2022-04-04T09:52:23.4159818Z grep error: pattern not found, got:
```
maybe we should just disable this on arch for now?

**Comment by Meulengracht** [comment]
> 

That is not the test this PR is fixing, I wanted to fix google:arch-linux-64:tests/main/snap-quota in another PR

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks

**Pull request merged**